dbCreate.sql

```diff
@@ -181,7 +181,7 @@ PRIMARY KEY (`NodeId`)
 
 CREATE TABLE IF NOT EXISTS `Users` (
 `UserId` int(10) unsigned NOT NULL AUTO_INCREMENT,
-`StudentCardId` int(10) unsigned NOT NULL,
+`StudentCardId` int(10) unsigned NULL,

```

Comment:
I don't think the NULL is necessary here, you could set the default to NULL. I'm not sure what effect, if any, this will have but might be worth testing before we merge.

Reply:
Having NULL here is necessary to satisfy the requirement that certain users will not have student cards. If inserting into users and NULL is passed as StudentCardId the foreign key will not fail. If a value other than NULL is provided the foreign key will need to be satisfied.

Reply:
Ah ok cool, that makes sense.